base/uk.ac.stfc.isis.ibex.instrument/src/uk/ac/stfc/isis/ibex/instrument/list/InstrumentListUtils.java

```diff
@@ -79,6 +94,66 @@ public boolean apply(InstrumentInfo item) {
         return returnValue;
     }
 
+
+    /**
+     * Given an observable for a list of instruments, filter out invalid
+     * instruments.
+     * 
+     * @param instruments instruments to filter
+     * @param logger a logger to log information and warnings about invalid
+     *            instruments
+     * 
+     * @return the valid instruments extracted from the input observable
+     */
+    public static Collection<InstrumentInfo> applyInstWhitelist(
+            Collection<InstrumentInfo> instruments,
+            Optional<ArrayList<String>> whitelist) {
+    	
+        Iterable<InstrumentInfo> filteredInstruments = Iterables.filter(instruments, new Predicate<InstrumentInfo>() {
+            @Override
+            public boolean apply(InstrumentInfo item) {
+                boolean instrumentWhitelisted;
+                
+                if (!whitelist.isPresent()) {
+                	instrumentWhitelisted = true;
+                } else if (whitelist.get().size() == 0) {
```

Comment:
Not sure if this check is needed if whitelist isn't available the check above should catch it